cellx/augmentation/image.py

```diff
@@ -63,3 +64,71 @@ def augment_random_flip(x: tf.Tensor) -> tf.Tensor:
     x = tf.image.random_flip_left_right(x)
     x = tf.image.random_flip_up_down(x)
     return x
+
+
+@augmentation_label_handler
+def augment_random_noise(x: tf.Tensor, stddev: float = 0.5) -> tf.Tensor:
+    """Perform a noise addition augmentation.
+
+    Parameters
+    ----------
+    x : tf.Tensor (T, ...)
+        The tensor to be augmented.
+
+    Returns
+    -------
+    x : tf.Tensor
+        The augmented tensor.
+    """
+    x_add = tf.random.normal(shape=tf.shape(x), stddev=stddev)
+    x = tf.add(x, x_add)
+    return x
+
+
+@augmentation_label_handler
+def augment_random_translation(x: tf.Tensor) -> tf.Tensor:
+    """Perform a random translation augmentation.
+
+    The image contents are shifted by a random x/y shift. The empty space left
+    behind is filled with zeros, while the image contents that are shifted out
+    of the field of view are discarded.
+
+    Parameters
+    ----------
+    x : tf.Tensor (T, ...)
+        The tensor to be augmented.
+
+    Returns
+    -------
+    x : tf.Tensor
+        The augmented tensor.
+    """
+    w, h = tf.shape(x)[-3], tf.shape(x)[-2]
+    dx = tf.random.uniform(
+        shape=(), minval=-w // 2, maxval=w // 2, dtype=tf.int32
+    )
+    dy = tf.random.uniform(
+        shape=(), minval=-h // 2, maxval=h // 2, dtype=tf.int32
+    )
+    x = tfa.image.translate(x, [dx, dy])
+    return x
+
+
+@augmentation_label_handler
+def augment_random_rot(x: tf.Tensor) -> tf.Tensor:
```

Comment:
Might be nice to parameterise this at some point. For example, exposing the angular range and interpolation options.